flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -507,20 +510,27 @@ public boolean isFinished() {
 	private Optional<InputWithData<InputChannel, BufferAndAvailability>> waitAndGetNextData(boolean blocking)
 			throws IOException, InterruptedException {
 		while (true) {
-			Optional<InputChannel> inputChannel = getChannel(blocking);
-			if (!inputChannel.isPresent()) {
+			Optional<InputChannel> optionalInputChannel = getChannel(blocking);
```

Comment:
also irrelated changes, should revert.

Reply:
name conflicts with the flowing InputChannel